File: app/src/main/java/com/mdgd/pokemon/models/repo/schemas/Move.java
package com.mdgd.pokemon.models.repo.schemas;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class Move {

    @SerializedName("move")
    @Expose
    private Move_ move;
    @SerializedName("version_group_details")
    @Expose
    private ArrayList<VersionGroupDetail> versionGroupDetails = new ArrayList<>();

    public Move_ getMove() {
        return move;
    }

    public void setMove(Move_ move) {
        this.move = move;
    }

    public ArrayList<VersionGroupDetail> getVersionGroupDetails() {
        return versionGroupDetails;
    }

    public void setVersionGroupDetails(ArrayList<VersionGroupDetail> versionGroupDetails) {
        this.versionGroupDetails = versionGroupDetails;
    }

}

File: models_impl/src/main/java/com/mdgd/pokemon/models_impl/repo/schemas/Move.kt
package com.mdgd.pokemon.models_impl.repo.schemas

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class Move {
    @Expose
    @SerializedName("move")
    var move: Move_? = null

    @Expose
    @SerializedName("version_group_details")
    var versionGroupDetails: List<VersionGroupDetail> = ArrayList()
}
